packages/shade_protocol/src/dex.rs

```diff
@@ -36,59 +36,69 @@ pub struct TradingPair {
 /* give_amount into give_pool
  * returns how much to be received from take_pool
  */
-pub fn pool_take_amount(give_amount: Uint128, give_pool: Uint128, take_pool: Uint128) -> Uint128 {
-    take_pool - ((give_pool * take_pool) / (give_pool + give_amount))
+
+pub fn pool_take_amount(give_amount: cosmwasm_std::Uint128, give_pool: cosmwasm_std::Uint128, take_pool: cosmwasm_std::Uint128) -> cosmwasm_std::Uint128 {
+    cosmwasm_std::Uint128(take_pool.u128() - give_pool.u128() * take_pool.u128() / (give_pool + give_amount).u128())
```

Comment:
Its better to use `cosmwasm_math_compat`'s checked math functions for added safety
```suggestion
take_pool.checked_sub(give_pool.checked_mul(take_pool)?.checked_div(give_pool.checked_add(give_amount)?)?)?.into()
```